pyomo/contrib/solver/solution.py

```diff
@@ -184,18 +204,28 @@ def get_primals(
     def get_duals(
         self, cons_to_load: Optional[Sequence[_GeneralConstraintData]] = None
     ) -> Dict[_GeneralConstraintData, float]:
+        if self._nl_info is None:
+            raise RuntimeError(
+                'Solution loader does not currently have a valid solution. Please '
+                'check the termination condition.'
+            )
+        if len(self._nl_info.eliminated_vars) > 0:
+            raise NotImplementedError('For now, turn presolve off (opt.config.writer_config.linear_presolve=False) to get dual variable values.')
+        assert self._sol_data is not None
```

Comment:
Should this be in here?

Reply:
Is there a reason not to?

Reply:
If the assertion is false and a user hits it, it's going to error without a helpful message. I don't have a problem with assertions, but I do have a problem with non-error-messages.

Reply:
fair. How is this?

Reply:
Better. I may tweak it a bit, but looks good!